### PR TITLE
Fix installation with specific branch, refs #148

### PR DESCRIPTION
**Pull request by poupotte**

### CODE REVIEW

File: src/application.coffee [hunk 1]
```diff
@@ -175,11 +176,14 @@ recoverManifest = (app, options, callback) ->
 
     else
         manifest.git = "https://github.com/cozy/cozy-#{app}.git"
-
-        # Check if application exists in market
-        homeClient.get 'api/applications/market', (err, res, market) ->
-            marketManifest = find market, (appli) -> appli.name is app
-            callback null, marketManifest or manifest
+        if options.branch
+            callback null, manifest
```
Comment (nono) [inline src/application.coffee]: @poupotte in that case, we don't check the market for the application icon. Is it normal?

File: src/application.coffee [hunk 2]
```diff
@@ -151,36 +151,40 @@ recoverManifest = (app, options, callback) ->
...
+            # Check if application exists in market
+            # if user doesn't specify branch or repository
             callback null, marketManifest or manifest
 
+        else
+            manifest.icon = marketManifest?.icon or ''
+
+            # Check if user has specified a branch
+            if options.branch
+                manifest.git = marketManifest.git
```
Comment (nono) [inline src/application.coffee]: @poupotte I'd have added `if marketManifest?.git` to this line, to preserve the default value for app that are not in the market

**Comment by frankrousseau** [comment]
@poupotte build is red.

**Comment by poupotte** [comment]
@frankrousseau : it's ok now, it was a travis issue :S

**Comment by nono** [comment]
:+1:

**Pull request merged**